### PR TITLE
Add nodeSelector to OVNKubeConfig CRD

### PR DESCRIPTION
**Pull request by pliurh**
Instead of using the same node label as CNO, we shall use a separate
label for MCO to avoid the interlock, as MCO needs the ovnkube CNI
running to drain a node.

### CODE REVIEW

**Comment by pliurh** [comment]
@zshi @Billy99 PTAL.

File: README.md [hunk 1]
```diff
@@ -79,15 +79,18 @@ locally for development purpose.
...
     ```
 
    1. `kubeConfigFile` stores the secret name of the tenant cluster kubeconfig
       file. The operator uses this to access the api-server of the tenant
       cluster.
    2. `poolName` specifies the name of the MachineConfigPool CR which contains
-      all the BF2 nodes in the infra cluster. The operator copies the
-      `spec.nodeSelector` of the MCP to render the ovnkube-node daemonset.
+      all the BF2 nodes in the infra cluster. 
+   3. `nodeSelector` The operator copies it to the `spec.nodeSelector` of MCP.
```
Comment (zshi-redhat) [inline README.md:93]: Also a note here indicating that the nodeSelector should not match with the CNO dpu label?

Comment (zshi-redhat) [reply README.md:93]: Ideally, if users create the MCO by themselves, this field should not be needed, right?

File: api/v1alpha1/ovnkubeconfig_types.go [hunk 2]
```diff
@@ -33,6 +33,8 @@ type OVNKubeConfigSpec struct {
 	// PoolName is the name of the MachineConfigPool CR which contains
 	// the BF2 nodes in the infra cluster.
 	PoolName string `json:"poolName"`
+	// nodeSelector specifies a label selector for Machines
```
Comment (zshi-redhat) [inline api/v1alpha1/ovnkubeconfig_types.go:36]: Would it make sense to have one upper layer structure for MCO config (for example, pool config), then include both PoolName and NodeSelector as sub options in it?
It doesn't seem straight forward to have PoolName or NodeSelector under OVNKubeConfig CRD w/o further explanation that these configs would be applied to MCO.

File: controllers/ovnkubeconfig_controller.go [hunk 3]
```diff
@@ -272,34 +271,36 @@ func (r *OVNKubeConfigReconciler) getLocalOvnkubeImage() (string, error) {
...
 			err = r.Create(context.TODO(), mcp)
 			if err != nil {
 				return fmt.Errorf("couldn't create MachineConfigPool: %v", err)
 			}
-			logger.Info("Created MachineConfigPool:", "name", mcpName)
+			logger.Info("Created MachineConfigPool:", "name", cs.PoolName)
 		}
 	} else {
-		if !(equality.Semantic.DeepEqual(foundMcp.Spec.MachineConfigSelector, mcSelector) && equality.Semantic.DeepEqual(foundMcp.Spec.NodeSelector, nodeSelector)) {
+		if !(equality.Semantic.DeepEqual(foundMcp.Spec.MachineConfigSelector, mcSelector) && equality.Semantic.DeepEqual(foundMcp.Spec.NodeSelector, cs.NodeSelector)) {
```
Comment (zshi-redhat) [inline controllers/ovnkubeconfig_controller.go:303]: Do we want to update MCO definition if user already creates it?

File: README.md (hunk 1 above)
Comment (pliurh) [reply README.md:93]: I don't think we shall let users create/modify the MCP if the DPU operator does that.

File: api/v1alpha1/ovnkubeconfig_types.go (hunk 2 above)
Comment (pliurh) [reply api/v1alpha1/ovnkubeconfig_types.go:36]: In the future, we could also support vanilla k8s. We can also use this API without mapping to an MCP.

File: config/crd/bases/dpu.openshift.io_ovnkubeconfigs.yaml [hunk 4]
```diff
@@ -40,6 +40,50 @@ spec:
...
+                  matchLabels:
+                    additionalProperties:
+                      type: string
+                    description: matchLabels is a map of {key,value} pairs. A single
+                      {key,value} in the matchLabels map is equivalent to an element
+                      of matchExpressions, whose key field is "key", the operator
+                      is "In", and the values array contains only "value". The requirements
+                      are ANDed.
+                    type: object
+                type: object
```
Comment (bn222) [inline config/crd/bases/dpu.openshift.io_ovnkubeconfigs.yaml:86]: I don't understand why we need this change twice. It looks like it is the same change as you propose for bundle/manifests/dpu.openshift.io_ovnkubeconfigs.yaml above.

File: manifests/stable/dpu.openshift.io_ovnkubeconfigs.yaml [hunk 5]
```diff
@@ -38,6 +38,50 @@ spec:
...
+                  matchLabels:
+                    additionalProperties:
+                      type: string
+                    description: matchLabels is a map of {key,value} pairs. A single
+                      {key,value} in the matchLabels map is equivalent to an element
+                      of matchExpressions, whose key field is "key", the operator
+                      is "In", and the values array contains only "value". The requirements
+                      are ANDed.
+                    type: object
+                type: object
```
Comment (bn222) [inline manifests/stable/dpu.openshift.io_ovnkubeconfigs.yaml:84]: Also looks like this is the same as above?

File: README.md (hunk 1 above)
Comment (pliurh) [reply README.md:93]: > What happens if the MCO is already created by the user?

Do you mean the MCP? If it already exists, the operator won't do anything to it. There is no reconciliation logic for the MCP. Due to some unknown reason, updating an MCP may cause the MCO malfunction.

File: config/crd/bases/dpu.openshift.io_ovnkubeconfigs.yaml (hunk 4 above)
Comment (pliurh) [reply config/crd/bases/dpu.openshift.io_ovnkubeconfigs.yaml:86]: This is part of the operator-sdk automation. When we need to update the CRD, we first need to update the go file in `api` fold, then run `make manifests` and `make generate` to generate the CRD manifests in `config\crd\` and other go code, at last run `make bundle` to generate the operator bundle manifests which will be used by OLM.

File: manifests/stable/dpu.openshift.io_ovnkubeconfigs.yaml (hunk 5 above)
Comment (pliurh) [reply manifests/stable/dpu.openshift.io_ovnkubeconfigs.yaml:84]: This is the operator bundle file for ART.

**Pull request merged**